src/go/rucio-dataset-mon-go/controllers/controllers.go

```diff
@@ -0,0 +1,244 @@
+func GetShortUrlParam() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		log.SetFlags(log.LstdFlags | log.Lshortfile)
+		verboseControllerInitLog(c)
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+
+		// Get request json with validation
+		req := models.ShortUrlRequest{}
+		if err := c.ShouldBindBodyWith(&req, binding.JSON); err != nil {
+			tempReqBody, _ := c.Get(gin.BodyBytesKey)
+			utils.ErrorResponse(c, "Bad request", err, string(tempReqBody.([]byte)))
+		}
+		defer cancel()
+		requestHash := short_url.GetShortUrl(ctx, c, req)
+		c.JSON(http.StatusOK,
+			requestHash,
+		)
+		verboseControllerOutLog(start, "GetShortUrlParam", req, requestHash)
+		return
+	}
+}
+
+// GetIndexPageFromShortUrlId controller that returns page from short url hash id
+func GetIndexPageFromShortUrlId() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		verboseControllerInitLog(c)
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+		defer cancel()
+
+		hashId := c.Param("id")
+		log.Printf("[INFO] Hash Id: %s", hashId)
+		shortUrlObj := short_url.GetRequestFromShortUrl(ctx, c, hashId)
+
+		dataTimestamp := data_timestamp.GetDataSourceTimestamp(ctx, c)
+		c.HTML(
+			http.StatusOK,
+			"index.html",
+			gin.H{
+				"title":                "Home Page",
+				"is_short_url":         "true",
+				"dt_request_short_url": shortUrlObj.Request,
+				"dt_saved_state":       shortUrlObj.SavedState,
+				"data_timestamp":       dataTimestamp.CreatedAt,
+			},
+		)
+		verboseControllerOutLog(start, "GetIndexPageFromShortUrlId", shortUrlObj, hashId)
+		return
+	}
+}
+
+// GetSingleDetailedDs controller that returns detailed dataset in TAPE or DISK
+func GetSingleDetailedDs() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		verboseControllerInitLog(c)
+		start := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+		defer cancel()
+
+		r := models.SingleDetailedDatasetsRequest{}
+		if err := c.ShouldBindBodyWith(&r, binding.JSON); err != nil {
+			x, _ := c.Get(gin.BodyBytesKey)
+			utils.ErrorResponse(c, "Bad request", err, string(x.([]byte)))
+			return
+		}
+
+		detailedRows := detailed_datasets.GetSingleDataset(ctx, c, r)
+		c.HTML(http.StatusOK,
+			"rse_detail_table.html",
+			gin.H{"data": detailedRows},
+		)
+		verboseControllerOutLog(start, "GetSingleDetailedDs", r, detailedRows)
+		return
+	}
+}
+
+// GetIndexPage serves index.html page
+func GetIndexPage(c *gin.Context) {
+	start := time.Now()
+	verboseControllerInitLog(c)
+	ctx, cancel := context.WithTimeout(context.Background(), mongo.Timeout)
+	defer cancel()
+
+	// get source data creation time
+	dataTimestamp := data_timestamp.GetDataSourceTimestamp(ctx, c)
+	c.HTML(
+		http.StatusOK,
+		"index.html",
+		gin.H{
+			"title":          "Home Page",
+			"is_short_url":   "false",
```

Comment:
why `"false"` is a string data-type instead of normal boolean?

Reply:
I put necessary comment for this. It should be `"false"` string since JavaScript function will behave it accordingly. 
`var isShortUrl = ({{.is_short_url}} === "true");`
I don't know if we can pass Go `false` value to JavaScript boolean type. Using string type is one of the secure ways I guess.